### PR TITLE
Code refactoring 

### PR DESCRIPTION
**Pull request by AntoniaSzecsi**

### CODE REVIEW

File: devices/Light.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include "Device.h"
+
+namespace SmartHome {
+    class Light : public Device {
+    public:
+        Light();
+        void setBrightness(int brightness);
+        int getBrightness() const;
+        void displayStatuses() const override;
+     Light* clone() const override;
```
Comment (CristianIonutIonescu) [inline devices/Light.h]: minor: indentation

File: user/User.cc [hunk 2]
```diff
@@ -1,135 +1,138 @@
...
+        for (auto device : other.devices) {
+            devices.push_back(device->clone());
         }
     }
 
+    // Utility function for clearing and moving devices
+    void User::moveDevices(User& other) {
+        devices = std::move(other.devices);
+        other.devices.clear();
+    }
```
Comment (CristianIonutIonescu) [inline user/User.cc:67]: not 100% this works as expected but we'll double check in the future presentations

**Review by CristianIonutIonescu — APPROVED** [review]

File: user/User.cc (hunk 2 above)
Comment (AntoniaSzecsi) [reply user/User.cc:67]: da, am observat acum ca nu e tocmai the best way. voi corecta si deschid PR, am observat si ca namespace-urile nu imi respecta naming conventions si le voi schimba in lowercases

**Review by CristianIonutIonescu — APPROVED** [review]
unique vs shared pointers will be learnt in the next presentation, no worries about it now.. you'll have time to think of which one should be used depending on context :)

hint: 95% of cases, unique_ptr is the correct usage :D

**Pull request merged**